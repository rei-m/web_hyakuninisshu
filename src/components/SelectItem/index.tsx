import * as React from 'react';
import { Field } from 'formik';
import styled from '@src/styles/styled-components';

export interface Props {
  title: string;
  name: string;
  value: string;
  valueList: Array<string | number>;
  nameList: string[];
  handleChange: (e: React.SyntheticEvent<HTMLSelectElement>) => void;
}

const Label = styled.label`
  text-align: left;
`;

const SelectRow = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 ${({ theme }) => theme.spacing2x};
`;

const SelectRange = styled.div`
  width: 100%;
`;

const SelectItem: React.FC<Props> = ({ title, name, value, handleChange, valueList, nameList }) => (
  <Label className="bp3-label">
    {title}
    <SelectRow>
      <SelectRange className="bp3-select bp3-large">
        <Field component="select" name={name} value={value} onChange={handleChange}>
          {valueList.map((v, i) => (
            <option value={v} key={`select_item_${i}`}>
              {nameList[i]}
            </option>
          ))}
        </Field>
      </SelectRange>
    </SelectRow>
  </Label>
);

export default SelectItem;
