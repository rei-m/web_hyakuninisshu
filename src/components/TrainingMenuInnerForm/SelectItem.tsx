import * as React from 'react';
import { Field } from 'formik';
import styled from 'styled-components';
import { withAppTheme } from '../../styles';

export interface SelectItemProps {
  readonly title: string;
  readonly name: string;
  readonly value: string;
  readonly valueList: Array<string | number>;
  readonly nameList: string[];
  readonly handleChange: (e: React.SyntheticEvent<HTMLSelectElement>) => void;
}

const Label = styled.label`
  text-align: left;
`;

const SelectRow = withAppTheme(styled.div)`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 ${({ theme }) => theme.spacing2x};
`;

const SelectRange = styled.div`
  width: 100%;
`;

const SelectItem = (props: SelectItemProps) => (
  <Label className="pt-label">
    {props.title}
    <SelectRow>
      <SelectRange className="pt-select pt-large">
        <Field
          component="select"
          name={props.name}
          value={props.value}
          onChange={props.handleChange}
        >
          {props.valueList.map((value, i) => (
            <option value={value} key={`select_item_${i}`}>
              {props.nameList[i]}
            </option>
          ))}
        </Field>
      </SelectRange>
    </SelectRow>
  </Label>
);

export default SelectItem;
