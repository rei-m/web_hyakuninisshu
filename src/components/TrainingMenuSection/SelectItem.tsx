import * as React from 'react';
import styled from 'styled-components';
import { OptionItem } from '../../types';

const Label = styled.label`
  text-align: left;
`;

const SelectRow = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 16px;
`;

const SelectRange = styled.div`
  width: 100%;
`;

export interface SelectItemProps {
  readonly title: string;
  readonly value: string | number;
  readonly list: Array<OptionItem<string | number>>;
  readonly onChange: (value: string) => void;
}

const onChange = (props: SelectItemProps) => {
  return (e: React.SyntheticEvent<HTMLSelectElement>) => {
    props.onChange(e.currentTarget.value);
  };
};

const SelectItem = (props: SelectItemProps) => (
  <Label className="pt-label">
    {props.title}
    <SelectRow>
      <SelectRange className="pt-select pt-large">
        <select value={props.value as any} onChange={onChange(props)}>
          {props.list.map((item, i) => (
            <option value={item.value as any} key={`select_item_${i}`}>
              {item.name}
            </option>
          ))}
        </select>
      </SelectRange>
    </SelectRow>
  </Label>
);

export default SelectItem;
