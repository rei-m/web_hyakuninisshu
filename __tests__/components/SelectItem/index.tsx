import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import SelectItem, { Props } from '@src/components/molecules/SelectItem';

describe('<SelectItem />', () => {
  it('should render component', () => {
    const props: Props = {
      title: 'title',
      name: 'name',
      value: 'value',
      valueList: [1, 2, 3],
      nameList: ['1', '2', '3'],
      handleChange: jest.fn(),
    };
    const renderer = ReactTestRenderer.create(<SelectItem {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
