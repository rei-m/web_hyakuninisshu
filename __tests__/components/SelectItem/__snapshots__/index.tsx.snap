// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectItem /> should render component 1`] = `
<div
  className="MuiFormControl-root-4 Component-formControl-1"
>
  <label
    className="MuiFormLabel-root-19 MuiFormLabel-filled-23 MuiInputLabel-root-8 MuiInputLabel-formControl-13 MuiInputLabel-animated-16 MuiInputLabel-shrink-15 Component-label-2"
    data-shrink={true}
    htmlFor="id-name"
  >
    title
  </label>
  <div
    className="MuiInputBase-root-46 MuiInput-root-33 MuiInput-underline-37 MuiInputBase-formControl-47 MuiInput-formControl-34 Component-select-3"
    onClick={[Function]}
  >
    <div
      className="MuiSelect-root-26"
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        className="MuiSelect-select-27 MuiSelect-selectMenu-30 MuiInputBase-input-56 MuiInput-input-41"
        id="select-name"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        id="id-name"
        name="name"
        type="hidden"
        value="value"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-63 MuiSelect-icon-32"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
